base/essentials.jl

```diff
@@ -601,6 +601,20 @@ end
 
 Val(x) = (@_pure_meta; Val{x}())
 
+eltype(::Val{T}) where T = T
+
+(&)(v1::Val{true}, v2::Val{true}) = Val{true}()
+(&)(v1::Val, v2::Val) = Val{false}()
```

Comment:
these methods seem to cover too many possibilities; I'd suggest to just write out all four combinations